### PR TITLE
Add write suppport

### PR DESCRIPTION
**Pull request by rhn**
Long story short, a writing module has been added with examples. Some changes were needed in handling sizes to be able to add new entries.

### CODE REVIEW

**Comment by andygrove** [comment]
Thanks @rhn .. could you fix the conflicts (I just merged some other PRs).

Also, are you interested in taking ownership of this crate? I no longer have the motivation to maintain this.

**Comment by rhn** [comment]
Rebased.

Why not. I can't promise long-term interest but I can merge things and maybe add features whenever I come back.

File: src/lib.rs [hunk 1]
```diff
@@ -1,9 +1,10 @@
 #![doc = include_str!("../README.md")]
 
+mod writer;
```
Comment (andygrove) [inline src/lib.rs:3]: Is there a `writer.rs` that needs to be added to this PR?

```
$ cargo test
   Compiling libc v0.2.153
   Compiling memmap2 v0.6.2
   Compiling riff-io v0.1.2 (/home/andy/git/personal/riff-io)
error[E0583]: file not found for module `writer`
 --> src/lib.rs:3:1
  |
3 | mod writer;
  | ^^^^^^^^^^^
  |
  = help: to create the module `writer`, create file "src/writer.rs" or "src/writer/mod.rs"
  = note: if there is a `mod writer` elsewhere in the crate already, import it with `use crate::...` instead
```

Comment (rhn) [reply src/lib.rs:3]: d'oh. Fixed.

**Review by andygrove — APPROVED** [review]
Thanks @rhn

**Comment by andygrove** [comment]
I published a new version to crates.io with this change

**Pull request merged**